capellambse_context_diagrams/collectors/exchanges.py

```diff
@@ -299,3 +299,19 @@ def collect(self) -> cabc.MutableSequence[_elkjs.ELKInputEdge]:
             )
 
         return self.data["edges"]
+
+
+def is_hierarchical(
+    ex: common.GenericElement,
+    box: _elkjs.ELKInputChild,
+    key: t.Literal["ports"] | t.Literal["children"] = "ports",
+) -> bool:
+    """Check if the exchange is hierarchical (nested) inside ``box``."""
+    is_parent = ex.parent.uuid == box["id"]
```

Comment:
We can't rely on the exchange's immediate parent being the box in question, it's legal for the exchange to sit anywhere in the model. This could happen e.g. if an existing exchange is reconnected to different ports. To make the check reliable, we need to look at the source and target instead, so something like:

    is_parent = src.parent.uuid == box["id"] == trg.parent.uuid

(Though I'm not sure how well this works in situations with/without ports.)